flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionTest.java

```diff
@@ -520,6 +519,47 @@ public void testEphemeralZooKeeperNodes() throws Exception {
 		}
 	}
 
+	@Test
+	public void testNotLeader() throws Exception {
+
+		final TestingLeaderElectionEventHandler electionEventHandler =
+			new TestingLeaderElectionEventHandler(TEST_LEADER);
+		final TestingLeaderRetrievalEventHandler retrievalEventHandler = new TestingLeaderRetrievalEventHandler();
+		LeaderElectionDriver leaderElectionDriver = null;
+		LeaderRetrievalDriver leaderRetrievalDriver = null;
+		try {
+
+			leaderElectionDriver =
+				ZooKeeperUtils.createLeaderElectionDriverFactory(client, configuration)
+					.createLeaderElectionDriver(electionEventHandler, electionEventHandler::handleError, TEST_URL);
+			electionEventHandler.init(leaderElectionDriver);
+
+			electionEventHandler.waitForLeader(timeout);
+			assertThat(electionEventHandler.getConfirmedLeaderInformation(), is(TEST_LEADER));
+
+			// Leader is revoked
+			((ZooKeeperLeaderElectionDriver) leaderElectionDriver).notLeader();
+			electionEventHandler.waitForRevokeLeader(timeout);
+			assertThat(electionEventHandler.getConfirmedLeaderInformation(), is(LeaderInformation.empty()));
+			// The data on ZooKeeper it not be cleared
```

Comment:
Since in the currently ZooKeeper HA implementation, check the leadership and then update the ZooKeeper node(aka clean up the leader information here) is not a transactional operation. So we may wrongly update the ZooKeeper node with empty data. What I have done is do nothing and the ephemeral node disappear automatically.

I add this test to verify this behavior and make sure when the behavior changed we could find it.